enterprise/server/remote_execution/blockio/blockio.go

```diff
@@ -337,18 +340,19 @@ func (c *COWStore) WriteAt(p []byte, off int64) (int, error) {
 	for len(p) > 0 {
 		// On each iteration, write to one chunk, first copying the readonly
 		// chunk if needed.
-		if !c.dirty[chunkOffset] {
-			// If we're newly dirtying a chunk, copy.
-			if err := c.copyChunk(chunkOffset); err != nil {
-				return 0, status.WrapError(err, "failed to copy chunk")
-			}
+		if err := c.copyChunk(chunkOffset); err != nil {
+			return 0, status.WrapError(err, "failed to copy chunk")
 		}
```

Comment:
imo the old code was slightly clearer - you could use the new `Dirty()` func to ensure `c.dirty` access is locked, or to keep the code less indented you could rename `copyChunk` to `copyChunkIfDirty`